go/test/endtoend/vreplication/cluster_test.go

```diff
@@ -624,12 +697,45 @@ func (vc *VitessCluster) AddShards(t *testing.T, cells []*Cell, keyspace *Keyspa
 
 		log.Infof("Finished creating shard %s", shard.Name)
 	}
+	for _, shard := range shardNames {
+		require.NoError(t, cluster.WaitForHealthyShard(vc.VtctldClient, keyspace.Name, shard))
+	}
+
+	waitTimeout := 30 * time.Second
+	vtgate := cells[0].Vtgates[0]
+	for _, shardName := range shardNames {
+		shard := keyspace.Shards[shardName]
+		numReplicas, numRDOnly := 0, 0
+		for _, tablet := range shard.Tablets {
+			switch strings.ToLower(tablet.Vttablet.TabletType) {
+			case "replica":
+				numReplicas++
+			case "rdonly":
+				numRDOnly++
+			}
+		}
+		numReplicas-- // account for primary, which also has replica type
```

Comment:
I think this is really about the fact that the shard hasn't elected a primary yet? The comment is not clear to me.

Reply:
When we wait for n replicas, the primary is not counted as one, even though the Vitess tablet type is `replica`.